tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -71,36 +81,74 @@ object ScatterGather {
     /**
       * Produces a new [[Scatter]] from the existing scatter.  Partitions the the scattered objects of type Result
       * according to the given discriminator function.  The discriminator function maps an individual object of type
-      * Result to a key of type K.  The partitions are each a sequence of objects of type Result that share the same key
-      * of type K when the discriminator function is applied.
+      * Result to a key of type Key.  The partitions are each a sequence of objects of type Result that share the same key
+      * of type Key when the discriminator function is applied.
       *
       * The resulting Scatter[(Key, Seq[Result])] can be further mapped or gathered (or both!).
       */
     def groupBy[Key](f: Result => Key) : Scatter[(Key, Seq[Result])]
-  }
 
+    /**
+      * Converts this [[Scatter]] of scattered scatterers into a [[Scatter]] formed by the elements of the scatterers.
+      * In other words, this scatter must be consist of elements that themselves are scatters of the resulting element
+      * type.
+      *
+      * @param toScatter an implicit conversion which asserts that the element type this scatter is a [[Scatter]].
+      * @tparam NextResult the type of element of each scattered [[Scatter]]
+      */
+    def flatten[NextResult](implicit toScatter: Result => /*<:<!!!*/ Scatter[NextResult]): Scatter[NextResult]
+  }
 
   /**
    * Represents a gather operation in a scatter/gather pipeline.
    */
   sealed trait Gather[Source, Result <: Task] extends Task {
     private[ScatterGather] var sources: Option[Seq[Source]] = None
+    private[ScatterGather] var result: Option[Result] = None
+
+    /**
+      * Produces a new [[Scatter]] from the result of the [[Gather]].
+      * @param toPartitioner the partitioner that will partition [[Result]] into one or more [[NextResult]]s.
+      * @tparam NextResult the new type of the scatter
+      */
+    def scatter[NextResult](toPartitioner: Result => Partitioner[NextResult]): Scatter[NextResult] = {
+      val sub = Scatter(new DelegatingPartitioner[Result, NextResult](this.result.get, toPartitioner))
+      this ==> sub
+      sub
+    }
   }
 
   /** Implementation of a Gather that performs a single All->One gather operation. */
   private class SingleGather[Source, Result <: Task](f: Seq[Source] => Result) extends Gather[Source,Result] {
     override def getTasks: Traversable[_ <: Task] = sources match {
       case None    => throw new IllegalStateException("Gather.getTasks called before sources populated.")
-      case Some(a) => Some(f(a))
+      case Some(a) => result = Some(f(a)); result
```

Comment:
I find this icky.  At a minimum it would be helpful to have `this.result = ...`.  Better would be to find a way to re-organize this so that the parent trait defines `getTasks` and then enforces that sub-classes define a method does the building and returns the result, so the `Gather` trait can catch it.